testing/waybar: new aport

z3ntu opened the pull request.

maxice8 commented on testing/fmt/APKBUILD:
```diff
@@ -0,0 +1,41 @@
+url="https://fmt.dev/latest/index.html"
+arch="all"
+license="BSD-2-Clause"
+depends=""
```
empty variables can be removed for brevity

maxice8 commented on testing/fmt/APKBUILD:
```diff
@@ -0,0 +1,41 @@
+license="BSD-2-Clause"
+depends=""
+depends_dev=""
+makedepends="$depends_dev"
```
makedepends can be removed

maxice8 commented on testing/fmt/APKBUILD:
```diff
@@ -0,0 +1,41 @@
+# Contributor:
+# Maintainer:
```
Aren't you willing to maintain them ?

maxice8 commented on testing/spdlog/APKBUILD:
```diff
@@ -8,7 +8,7 @@ options="!check" # Unable to run tests
 arch="all"
 license="MIT"
-makedepends="cmake"
+makedepends="cmake fmt-dev"
```
These changes require spdlog to have its pkgrel bumped

maxice8 commented on testing/waybar/APKBUILD:
```diff
@@ -0,0 +1,36 @@
+url="https://github.com/Alexays/Waybar/"
+arch="all"
+license="MIT"
+depends=""
```
empty variables can be removed

maxice8 commented on testing/waybar/APKBUILD:
```diff
@@ -0,0 +1,36 @@
+# Contributor:
+# Maintainer:
```
Same question about maintainership for fmt applies here

maxice8 commented on testing/waybar/APKBUILD:
```diff
@@ -0,0 +1,36 @@
+
+check() {
+	# Replace with proper check command(s)
+	:
```
This needs to be fixed or a !check with a small comment telling why there is no check enabled made.

maxice8 commented on testing/waybar/APKBUILD:
```diff
@@ -0,0 +1,30 @@
+url="https://github.com/Alexays/Waybar/"
+arch="all"
+license="MIT"
+makedepends="meson gtkmm3-dev jsoncpp-dev spdlog-dev fmt-dev libinput-dev libnl-dev pulseaudio-dev libmpdclient-dev libgudev-dev"
```
Can this line be cut at about 80 chars or less ?

maxice8 replied:
not outdated

z3ntu commented:
I'm actually not sure if the spdlog change worked because spdlog still installs the bundled fmt headers...

maxice8 commented:
> I'm actually not sure if the spdlog change worked because spdlog still installs the bundled fmt headers...

Can't they just be deleted ?

maxice8 commented:
`fmt` install the pkgconfig file to /usr/share, please make it install to /usr/lib

z3ntu commented:
Fixed. I still have to test this on a device and figure out what difference the spdlog commit does; but I think that's all for this PR. Thanks for the extremely quick reviews @maxice8 :)

z3ntu commented:
So the spdlog commit makes this difference in the built `spdlog-dev` package (the `spdlog` package stays the same):
```
│ ├── usr/lib/cmake/spdlog/spdlogTargets.cmake
│ │ @@ -59,15 +59,17 @@
│ │    set(_IMPORT_PREFIX "")
│ │  endif()
│ │  
│ │  # Create imported target spdlog::spdlog
│ │  add_library(spdlog::spdlog INTERFACE IMPORTED)
│ │  
│ │  set_target_properties(spdlog::spdlog PROPERTIES
│ │ +  INTERFACE_COMPILE_DEFINITIONS "SPDLOG_FMT_EXTERNAL"
│ │    INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
│ │ +  INTERFACE_LINK_LIBRARIES "fmt::fmt"
│ │  )
│ │  
│ │  if(CMAKE_VERSION VERSION_LESS 3.0.0)
│ │    message(FATAL_ERROR "This file relies on consumers using CMake 3.0.0 or greater.")
│ │  endif()
│ │  
│ │  # Load information for each installed configuration.
│ ├── usr/lib/cmake/spdlog/spdlogConfig.cmake
│ │ @@ -17,15 +17,15 @@
│ │  # * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
│ │  # * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
│ │  # * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
│ │  # * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
│ │  # * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
│ │  # *************************************************************************/
│ │  
│ │ -set(SPDLOG_FMT_EXTERNAL OFF)
│ │ +set(SPDLOG_FMT_EXTERNAL ON)
│ │  
│ │  include("${CMAKE_CURRENT_LIST_DIR}/spdlogTargets.cmake")
│ │  
│ │  if(SPDLOG_FMT_EXTERNAL)
│ │      include(CMakeFindDependencyMacro)
│ │      find_dependency(fmt CONFIG)
│ │  endif()
│ ├── usr/lib/cmake/spdlog/spdlogConfigVersion.cmake
│ │ @@ -36,15 +36,15 @@
│ │  # if the installed project requested no architecture check, don't perform the check
│ │  if("FALSE")
│ │    return()
│ │  endif()
│ │  
│ │  # if the installed or the using project don't have CMAKE_SIZEOF_VOID_P set, ignore it:
│ │  if("${CMAKE_SIZEOF_VOID_P}" STREQUAL "" OR "8" STREQUAL "")
│ │ -   return()
│ │ +  return()
│ │  endif()
│ │  
│ │  # check that the installed version has the same 32/64bit-ness as the one which is currently searching:
│ │  if(NOT CMAKE_SIZEOF_VOID_P STREQUAL "8")
│ │    math(EXPR installedBits "8 * 8")
│ │    set(PACKAGE_VERSION "${PACKAGE_VERSION} (${installedBits}bit)")
│ │    set(PACKAGE_VERSION_UNSUITABLE TRUE)
```

maxice8 commented:
So i think  the bundled stuff can be removed from include/

anyways ping me here or on IRC and i'll merge this

z3ntu commented:
@maxice8 I've removed the spdlog patch for now as I've written the Arch Linux spdlog maintainer why he removed the patch which was deleting the bundled code from spdlog. I'll open a PR later if it makes any sense.

And I've added `ttf-font-awesome` as dependency for waybar as it uses that font for all icons in the bar.

This should be good to go now!
